MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/SnakingDirectionsRouteActivity.java

```diff
@@ -152,57 +163,74 @@ private void getDirectionsRoute(Point origin, Point destination) {
       .accessToken(getString(R.string.access_token))
       .build();
 
-    client.enqueueCall(new Callback<DirectionsResponse>() {
+    mapboxDirectionsClient.enqueueCall(new Callback<DirectionsResponse>() {
       @Override
       public void onResponse(Call<DirectionsResponse> call, Response<DirectionsResponse> response) {
-
         // Create log messages in case no response or routes are present
         if (response.body() == null) {
-          Timber.d("No routes found, make sure you set the right user and access token.");
+          Timber.d( "No routes found, make sure you set the right user and access token.");
           return;
         } else if (response.body().routes().size() < 1) {
-          Timber.d("No routes found");
+          Timber.d( "No routes found");
           return;
         }
 
-        // Get the route from the Mapbox Directions API
+        // Get the route from the Mapbox Directions API response
         DirectionsRoute currentRoute = response.body().routes().get(0);
-        List<Point> directionsPointsForLineLayer = new ArrayList<>();
-        for (int i = 0; i < currentRoute.legs().size(); i++) {
-          RouteLeg leg = currentRoute.legs().get(i);
-          List<LegStep> steps = leg.steps();
-          for (int j = 0; j < steps.size(); j++) {
-            LegStep step = steps.get(j);
-            List<StepIntersection> intersections = step.intersections();
-            for (int k = 0; k < intersections.size(); k++) {
-              Point location = intersections.get(k).location();
-              directionsPointsForLineLayer.add(Point.fromLngLat(location.longitude(), location.latitude()));
-              List<Feature> drivingRoutePolyLineFeatureList = new ArrayList<>();
-              LineString lineString = LineString.fromLngLats(directionsPointsForLineLayer);
-              List<Point> coordinates = lineString.coordinates();
-              for (int x = 0; x < coordinates.size(); x++) {
-                drivingRoutePolyLineFeatureList.add(Feature.fromGeometry(LineString.fromLngLats(coordinates)));
-              }
-
-              // Update the GeoJSON source
-              GeoJsonSource source = map.getStyle().getSourceAs(DRIVING_ROUTE_POLYLINE_SOURCE_ID);
-              if (source != null) {
-                source.setGeoJson(FeatureCollection.fromFeatures(drivingRoutePolyLineFeatureList));
-              }
-            }
-          }
-        }
+
+        // Start the step-by-step process of drawing the route
+        runnable = new DrawRouteRunnable(mapboxMap, currentRoute.legs().get(0).steps(), handler = new Handler());
```

Comment:
handler can be created inline in the field deceleration

Reply:
Fixed to `private Handler handler = new Handler();` at top of the activity. If I misunderstood your suggestion, let me know.